File: OdontoFlowAPI/OdontFlow/OdontFlow.Domain/DTOs/Order/CreateOrderDTO.cs
namespace OdontFlow.Domain.DTOs.Order;

public class CreateOrderDTO
{
    public Guid ClientId { get; set; }
    public string PatientName { get; set; } = default!;
    public string RequesterName { get; set; } = default!;
    public int OrderTypeId { get; set; }
    public DateTime CommitmentDate { get; set; }
    public string? Color { get; set; }
    public bool ApplyInvoice { get; set; }
    public string? Others { get; set; }
    public bool Bite { get; set; }
    public bool Models { get; set; }
    public bool Casts { get; set; }
    public bool Spoons { get; set; }
    public bool Attachments { get; set; }
    public bool Analogs { get; set; }
    public bool Screws { get; set; }
    public bool MetalArticulator { get; set; }
    public bool DisposableArticulator { get; set; }
    public List<CreateOrderItemDTO> Items { get; set; } = new();
    public string? Observations { get; set; }
    public string? CollectionNotes { get; set; }
    public string? DeliveryNotes { get; set; }

    public bool ? Uncollectible { get; set; }
}

File: OdontoFlowWeb/OdontFlow/src/app/core/model/create-order.dto.ts
import { CreateOrderItemDto } from "./order-item.model";

export interface CreateOrderDto {
    clientId?: string;  
    workGroupId?: number;
    orderStatusId?: number;
    orderTypeId?: number;
    colorId?: number;
  
    requesterName?: string;
    patientName?: string;
  
    bite?: boolean;
    models?: boolean;
    casts?: boolean;
    spoons?: boolean;
    attachments?: boolean;
    analogs?: boolean;
    screws?: boolean;
  
    others?: string;
    observations?: string;
  
    entryDate?: string;
    deliveryDate?: string;
    commitmentDate?: string;
  
    metalArticulator?: boolean;
    disposableArticulator?: boolean;
  
    cost?: number;
    balance?: number;
    payment?: number;
    tax?: number;
  
    collectionNotes?: string;
    deliveryNotes?: string;
  
    previousOrderId?: string;
    color?: string;
  
    priceListId?: number;
    uncollectible?: boolean;
    applyInvoice?: boolean;
    invoiceNumber?: string;
 
  
    items?: CreateOrderItemDto[];
  }
  